### PR TITLE
fix: validations array is not cleared

### LINKED ISSUE #103: violations are not cleared at the beginning of validate()
Following code was working in 1.x but fails in 2.x because `validations` array is not cleared:

```php

$validator = new Validator();

if ($validator->validate(/* invalid iban */) || $validator->validate(/* valid iban */)) {
  echo "One of the ibans is correct";
}
```

### PR DESCRIPTION
**Pull request by jacekkarczmarczyk**
fixes #103

### CODE REVIEW

File: tests/ValidatorTest.php [hunk 1]
```diff
@@ -245,4 +245,13 @@ public function test_iban_validation_results_in_InvalidChecksumException(): void
 
         $this->validator->validate(new Iban('DE90 3704 0044 0532 0130 00'), throw: true);
     }
+    
+    public function test_validations_are_cleared_before_validation(): void
+    {
+        $this->validator->validate(new Iban('DE89 3704 0044 053A 013B 00'));
+
```
Comment (jschaedl) [inline tests/ValidatorTest.php:252]: Let's add one more assertion:

```suggestion
$this->assertFalse($isValid);
```

**Review by jschaedl — CHANGES_REQUESTED** [review]
Only one small thing, but otherwise 👍🏻

File: tests/ValidatorTest.php (hunk 1 above)
Comment (jacekkarczmarczyk) [reply tests/ValidatorTest.php:254]: done

**Review by jschaedl — APPROVED** [review]

File: tests/ValidatorTest.php [hunk 2]
```diff
@@ -245,4 +245,15 @@ public function test_iban_validation_results_in_InvalidChecksumException(): void
 
         $this->validator->validate(new Iban('DE90 3704 0044 0532 0130 00'), throw: true);
     }
+    
+    public function test_validations_are_cleared_before_validation(): void
```
Comment (jschaedl) [inline tests/ValidatorTest.php:249]:
```suggestion
    public function test_validations_are_cleared_before_next_validation(): void
```

File: tests/ValidatorTest.php [hunk 3]
```diff
@@ -245,4 +245,15 @@ public function test_iban_validation_results_in_InvalidChecksumException(): void
...
+    {
+        $isValid = $this->validator->validate(new Iban('DE89 3704 0044 053A 013B 00'));
+
+        $this->assertFalse($isValid);
```
Comment (jschaedl) [inline tests/ValidatorTest.php:253]:
```suggestion
        self::assertFalse($isValid);
```

File: tests/ValidatorTest.php [hunk 4]
```diff
@@ -245,4 +245,15 @@ public function test_iban_validation_results_in_InvalidChecksumException(): void
...
+
+        $isValid = $this->validator->validate(new Iban('AD1200012030200359100100'));
+
+        $this->assertTrue($isValid);
```
Comment (jschaedl) [inline tests/ValidatorTest.php:257]:
```suggestion
        self::assertTrue($isValid);
```

File: src/Validator.php [hunk 5]
```diff
@@ -50,6 +50,8 @@ public function validate(string|Iban $iban, bool $throw = false): bool
         if (!$iban instanceof Iban) {
             $iban = new Iban($iban);
         }
+
```
Comment (jschaedl) [inline src/Validator.php:53]:
```suggestion

```

File: tests/ValidatorTest.php [hunk 6]
```diff
@@ -245,4 +245,15 @@ public function test_iban_validation_results_in_InvalidChecksumException(): void
 
         $this->validator->validate(new Iban('DE90 3704 0044 0532 0130 00'), throw: true);
     }
+
```
Comment (jschaedl) [inline tests/ValidatorTest.php:248]:
```suggestion

```

File: tests/ValidatorTest.php [hunk 7]
```diff
@@ -245,4 +245,14 @@ public function test_iban_validation_results_in_InvalidChecksumException(): void
 
         $this->validator->validate(new Iban('DE90 3704 0044 0532 0130 00'), throw: true);
     }
+    public function test_validations_are_cleared_before_next_validation(): void
```
Comment (jschaedl) [inline tests/ValidatorTest.php:248]:
```suggestion

    public function test_validations_are_cleared_before_next_validation(): void
```

**Comment by jschaedl** [comment]
Thank you @jacekkarczmarczyk !

**Pull request merged**